### ISSUE TITLE
Reordering of foreign key constraints

### ISSUE THREAD

**Issue by robarchibald** [ariga/atlas#1694]
This feels like a pretty important bug. Today I was comparing changes I had made using my current migration tool `sqitch` and comparing with the migration I was getting with `atlas`. I modified my `schema.hcl` and did a simple migrate diff like so: 
```bash
atlas migrate diff reportView \
  --dir file://migrations \
  --to file://../schema.hcl \
  --dev-url docker://postgres/15
```

I was perplexed when the columns I had in my multi-column index was out of order with `atlas`. I pored over the HCL file to see if I'd done something wrong and finally thought I'd try it again. I did and was perplexed to find that now a foreign key was now out of order. 

Is it possible that `atlas` is using Go maps to store the columns in indexes and foreign keys? If so, that'd explain why items can be out of order. And, in case you don't believe that ordering of columns matter, here's an article: https://www.enterprisedb.com/blog/index-order-does-matter that explains why ordering matters in indexes and keys.

**Comment by robarchibald** [comment]
I should note that the output order of the `migrate diff` is also not deterministic. Yet another hint that there are probably Go maps being used here. I think it would be much more helpful to be deterministic so that it's easier to sift through changes.

**Comment by robarchibald** [comment]
> Can you provide a minimal example to reproduce this issue? It will be a great help.

I'll see what I can do

**Comment by a8m** [comment]
Hey, @robarchibald 👋🏻
Can you check this on the latest canary version? I think this was resolved in https://github.com/ariga/atlas/issues/1614

**Comment by robarchibald** [comment]
@a8m, yes this appears to be fixed by #1614. Thanks
